### DISCUSSION TITLE
```Image.Load<TPixel>(imageSource.Invoke(), out IImageFormat imgFormat);``` Alternative

### DISCUSSION THREAD

**Discussion by TonyValenti** [SixLabors/ImageSharp#2557] [Q&A] [answered]
I currently have this code:
```
var image = Image.Load<TPixel>(imageSource.Invoke(), out IImageFormat imgFormat);
```

In the latest version of ImageSharp, this method has gone away.  

What is the right way for me to determine the image format of the image?

Specifically, I only need to know if the loaded image is transparent or not.

**Comment by tocsoft** [comment, marked as answer]
we got rid of the out overloads, that information is now available in the images metadata `image.Metadata.DecodedImageFormat` if you need both the image and the format.

If you just want the format that can be done using the `Image.DetectFormat(...)`/`Image.DetectFormatAsync(...)` APIs

or if you can call `Image.Identify(..)` / `Image.IdentifyAsync(..)` too if you want the metadata also, then like `Image.Load..(..)` you access it from `foo.Metadata.DecodedImageFormat`

This change was introduced to allow both our async and sync apis to have and identical signatures to normalise the patterns you have to follow to get those details.

**Reply by TonyValenti** [reply]
Thanks so much!

Yes.  Out sucks.  Definitely prefer this method.
